addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVWebSocket.java

```diff
@@ -0,0 +1,386 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketClose;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ValloxMVWebSocket} is responsible for socket communication with the vallox ventilation unit
+ *
+ * @author Björn Brings - Initial contribution
+ */
+public class ValloxMVWebSocket {
+    private String ip;
+    private ValloxMVHandler voHandler;
+    private WebSocketClient client;
+    private ValloxMVWebSocketListener socket;
+
+    private final Logger logger = LoggerFactory.getLogger(ValloxMVWebSocket.class);
+
+    public ValloxMVWebSocket(ValloxMVHandler voHandler, String ip) {
+        this.voHandler = voHandler;
+        this.ip = ip;
+        client = new WebSocketClient();
+    }
+
+    public void request(ChannelUID channelUID, String updateState) {
+        try {
+            socket = new ValloxMVWebSocketListener(channelUID, updateState);
+            client.start();
+
+            URI destUri = new URI("ws://" + ip + ":80");
+
+            ClientUpgradeRequest request = new ClientUpgradeRequest();
+            logger.debug("Connecting to: {}", destUri);
+            client.connect(socket, destUri, request);
+            socket.awaitClose(2, TimeUnit.SECONDS);
+        } catch (IOException e) {
+            connectionError(e);
+        } catch (Exception e) {
+            logger.error("Error: {}", e.getMessage());
+        }
+    }
+
+    public void close() {
+        try {
+            client.stop();
+        } catch (IOException e) {
+            connectionError(e);
+        } catch (Exception e) {
+            logger.error("Error: {}", e.getMessage());
+        }
+        client.destroy();
+        client = null;
+    }
+
+    public void connectionError(IOException e) {
+        logger.error("Error connecting vallox unit.", e);
```

Comment:
If this is caused by an `IOException` and you also update the state, debug is really enough for this, see also:

In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action.